include/boost/geometry/algorithms/detail/disjoint/linear_areal.hpp

```diff
@@ -98,6 +115,66 @@ struct disjoint_no_intersections_policy<Geometry1, Geometry2, Tag1, multi_tag>
         }
         return true;
     }
+
+    typedef typename point_type<Geometry1>::type point1_type;
+
+    template <typename Strategy>
+    static inline bool apply(Geometry1 const& g1,
+                             Geometry2 const& g2,
+                             Strategy const& strategy,
+                             point1_type& p)
+    {
+        // TODO: use partition or rtree on g2
+        typedef typename boost::range_iterator<Geometry1 const>::type iterator;
+        for ( iterator it = boost::begin(g1) ; it != boost::end(g1) ; ++it )
+        {
+            typedef typename boost::range_value<Geometry1 const>::type value_type;
+            if ( ! disjoint_no_intersections_policy<value_type const, Geometry2>
+                    ::apply(*it, g2, strategy, p) )
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+};
+
+template<typename Geometry1, typename Geometry2,
+         typename NoIntersectionsPolicy
+                    = disjoint_no_intersections_policy
+                      <Geometry1, Geometry2> >
+struct disjoint_linear_areal_with_info
+{
+    typedef typename point_type<Geometry1>::type p_type;
+
+    typedef segment_intersection_points<p_type> intersection_return_type;
+
+    template <typename Strategy>
+    static inline intersection_return_type
+    apply(Geometry1 const& g1, Geometry2 const& g2, Strategy const& strategy)
+    {
+        intersection_return_type res =
+        disjoint_linear_with_info<Geometry1, Geometry2>::apply(g1, g2, strategy);
```

Comment:
I'm not sure about it but shouldn't lines such as this be indented?

Reply:
From the wiki
```
The preferred line length is 80 characters, with maximum length of 100.
```

Reply:
I mean either:
```
intersection_return_type res =
    disjoint_linear_with_info<Geometry1, Geometry2>::apply(g1, g2, strategy);
```
or if that's not possible then:
```
intersection_return_type res = disjoint_linear_with_info
                                    <
                                        Geometry1, Geometry2
                                    >::apply(g1, g2, strategy);
```

Reply:
converting to 
```
intersection_return_type res = disjoint_linear_with_info
                                           <
                                               Geometry1,
                                               Geometry2
                                           >::apply(g1, g2, strategy);
```
since template parameters should be in different lines in this case.